siatest/renter/renter_test.go

```diff
@@ -569,3 +570,148 @@ func testRenterDownloadAfterRenew(t *testing.T, tg *siatest.TestGroup) {
 		t.Fatal(err)
 	}
 }
+
+// TestRenterSpendingReporting checks the accuracy for the reported
+// spending
+func TestRenterSpendingReporting(t *testing.T) {
+	// Create a group for the subtests
+	groupParams := siatest.GroupParams{
+		Hosts:  1,
+		Miners: 1,
+	}
+	tg, err := siatest.NewGroupFromTemplate(groupParams)
+	if err != nil {
+		t.Fatal("Failed to create group: ", err)
+	}
+	defer func() {
+		if err := tg.Close(); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	// Create Renter
+	renterDir, err := siatest.TestDir(t.Name(), "renter")
+	if err != nil {
+		t.Fatal(err)
+	}
+	r, err := siatest.NewCleanNode(node.Renter(renterDir))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Call Gateway
+	m := tg.Miners()[0]
+	err = r.GatewayConnectPost(m.GatewayAddress())
+	if err != nil {
+		t.Fatal(err)
+	}
+	mwg, err := m.WalletGet()
+	if err != nil {
+		t.Fatal(err)
+	}
+	rwag, err := r.WalletAddressGet()
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = m.WalletSiacoinsPost(mwg.ConfirmedSiacoinBalance.Div64(2), rwag.Address)
```

Comment:
It's not a blocking thing, but the excessive use of short/truncated variable names in this test are impacting the readability. Generally we try to keep it to only a few truncated names per function. The real issue here isn't any particular use, it's just that altogether there are too many.

A good rule of thumb is that the further away from creation/instantiation that you use a variable, the longer / more complete the variable name should be. A similar rule of thumb is that you shouldn't have more than 1 or 2 truncated variables being used in the same scope. That's all just to improve readability.